src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.hpp

```diff
@@ -21,10 +21,19 @@ static std::shared_ptr<dnnl::convolution_forward::primitive_desc> get_convolutio
     dnnl::memory::dims pad_l(prim->pad.begin(), prim->pad.end());
     dnnl::memory::dims pad_r(prim->pad.begin(), prim->pad.end());
 
+    auto grouped_weights = format::is_grouped(weights_layout.format) || prim->grouped_weights_shape;
+    if (grouped_weights && (input_layout.get_rank() == weights_layout.get_rank())) {
+        auto tensor = weights_layout.get_tensor();
+        if (tensor.spatial[0] == 1 && tensor.spatial[1] != 1) {
```

Comment:
Two questions,
1. How about the opposite case? (For example spatial[0] != 1 && spatial[1] == 1)
2. What about more than 4 dims?

Reply:
XY of weight tensor is changed by ngraph or something.
I'm not sure there is the case that used Y not X. If it is, this logic needs to know why XY values are swapped.

Tensor of group weight will be increased 1 dim more than input data.